src/models/users/user.py

```diff
@@ -0,0 +1,15 @@
+import uuid
+
+from src.models.tasks.task import Task
+
+
+class User(object):
+    def __init__(self, email, password, name, _id=None):
+        self.email = email
+        self.password = password
+        self.name = name
+        self._id = uuid.uuid4().hex if _id is None else _id
+
+    def add_new_task(self, title, description, due_date, folder):
+        new_task = Task(title, description, due_date, folder, self._id)
```

Comment:
Yeah the changes you suggest sound good. I think that makes more sense to
just have the folder_id stored for each task.

On Sat, Dec 19, 2015 at 12:23 AM Ibrahim MOKDAD <notifications@github.com>
wrote:

> In src/models/users/user.py
> <https://github.com/ibininja/soc_todo/pull/1#discussion_r48090262>:
>
> > @@ -0,0 +1,15 @@
> > +import uuid
> > +
> > +from src.models.tasks.task import Task
> > +
> > +
> > +class User(object):
> > +    def __init__(self, email, password, name, _id=None):
> > +        self.email = email
> > +        self.password = password
> > +        self.name = name
> > +        self._id = uuid.uuid4().hex if _id is None else _id
> > +
> > +    def add_new_task(self, title, description, due_date, folder):
> > +        new_task = Task(title, description, due_date, folder, self._id)
>
> I suggest the following:
>
>    - Tasks be created from within the folder class. So in Users class,
>    add_task would be as :
>    folder=get_folder_by_id(folder_id) folder.add_task(title, description)
>    #the user_id and folder_id would obtained from self.user_id and self._id
>    The reason is to make easier to know that tasks are only created if a
>    folder exist. If you agree to this point I will push the change since I
>    made it in my local branch you don't have to make any amends on this one
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ibininja/soc_todo/pull/1/files#r48090262>.
>